shell_elvish.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"strings"
+)
+
+type elvish struct {
+	re *strings.Replacer
+}
+
+var ELVISH = elvish{}
+
+func (elvish) Hook() (string, error) {
+	return `fn -direnv-hook {
+	direnv export elvish > .direnv.elv
+	-source .direnv.elv
+	/bin/rm .direnv.elv
+}
+
+edit:before-readline = [{ -direnv-hook }]
+`, nil
+}
+
+func (sh elvish) Export(e ShellExport) string {
+	out := ""
+	for k, v := range e {
+		switch v {
+		case nil:
+			out += sh.unset(k)
+		default:
+			out += sh.export(k, *v)
+		}
+	}
+	return out
+}
+
+func (sh elvish) export(k, v string) string {
+	return "E:" + k + " = \"" + sh.escape(v) + "\";\n"
```

Comment:
is the `\n` required or just nice to have for the formatting?

Reply:
I've found it was easier to debug things and/or inspect the output of `direnv export`